### DISCUSSION TITLE
How to hold state with useState in server side too?

### DISCUSSION THREAD

**Discussion by rslonik** [nuxt/nuxt#17696] [Questions]
I want to login a user and hold the user info in the state. I am using `/server/api` to authenticate and h3's setCookie to have the user jwt tokens written to a cookie.

I've tried to call the user endpoint in a composable or in a plugin, but both as I set the user with useState ends up in a "Hydration text mismatch" when user is set.

1. It's possible to setup a server only plugin with useState, but I couldn't make api (server/api) calls in it, is it possible?
2. Is there a way to set useState in server middleware?

**Comment by Norel** [comment]
> 1. It's possible to setup a server only plugin with useState, but I couldn't make api (server/api) calls in it, is it possible?

If you're using $fetch (ohmyfetch) you can add the baseURL parameter with the hostname. That fixed it for me.

> 2. Is there a way to set useState in server middleware?

This one is really important. Same with the server api. We need an access to the state for various reasons (auth for example).

**Reply by chenyuncai** [reply]
Q2: For now you can set the data to event.context.xxxData = {} in server/middleware or server/api , and then get the data in a global route middleware by useNuxtApp().ssrContext.event.context.xxxData, and set it to a useState.

**Comment by thejspro** [comment]
Any updates on this?
